### PR TITLE
MULE-18094: Pooled Platform Managed Connection Providers fail to be created

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/ocs/PlatformManagedOAuthConnectionProvider.java [hunk 1]
```diff
@@ -355,4 +360,44 @@ public ConnectionManagementType getConnectionManagementType() {
...
+        public void onReturn(C connection) {
+          ((PoolingListener) unwrappedDelegate).onReturn(connection);
+        }
+      };
+    } else {
+      return new PoolingListener<C>() {
+
+        @Override
+        public void onBorrow(C connection) {
+          throw new UnsupportedOperationException("The delegate connection provider does not support Pooling");
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/ocs/PlatformManagedOAuthConnectionProvider.java]: This breaks liskov principle. You should use the NullObject pattern here instead

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
Validation job finished successfully, yet the validation is still pending, I am triggering the job again.

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Pull request merged**